src/peft/utils/incremental_pca.py

```diff
@@ -66,11 +68,18 @@ def __init__(
             if lowrank_q < n_components:
                 raise ValueError("lowrank_q must be greater than or equal to n_components.")
 
-            def svd_fn(X):
-                U, S, V = torch.svd_lowrank(X, q=lowrank_q, niter=lowrank_niter)
+            def svd_fn(X, seed):
```

Comment:
Would it make sense to turn `svd_fn` into a proper method, with all arguments stored as `self` attributes (so `self.lowrank = lowrank` etc.). Then we don't need to `partial` over the arguments. Furthermore, this is more in line with sklearn: Save all the class arguments as `self` attributes and don't put any further logic inside of `__init__` -- this is because this logic is not executed when running `set_params` on the object.

Of course, this `IncrementalPCA` class is not a proper sklearn `BaseEstimator` with `get_params` and `set_params`, so it doesn't really matter, but it could still be nice for possible future integration to keep with the sklearn standards.

Reply:
done, I cleaned up the __init__ method and defined the svd methods as proper class methods